ssl/ssl_lib.cc

```diff
@@ -1759,13 +1763,58 @@ int SSL_get_extms_support(const SSL *ssl) {
   return 0;
 }
 
-int SSL_CTX_get_read_ahead(const SSL_CTX *ctx) { return 0; }
+int SSL_CTX_get_read_ahead(const SSL_CTX *ctx) {
+  return ctx->enable_read_ahead;
+}
+
+int SSL_get_read_ahead(const SSL *ssl) {
+  return ssl->enable_read_ahead;
+}
 
-int SSL_get_read_ahead(const SSL *ssl) { return 0; }
+void SSL_CTX_set_default_read_buffer_len(SSL_CTX *ctx, size_t len) {
+  // SSLBUFFER_MAX_CAPACITY(0xffff) is the maximum SSLBuffer supports reading at one time
+  if (len > SSLBUFFER_MAX_CAPACITY) {
+    len = SSLBUFFER_MAX_CAPACITY;
+  }
+  // Setting a very small read buffer won't cause issue because the SSLBuffer
+  // will always read at least the amount of data specified in the TLS record
+  // header
+  ctx->read_ahead_buffer_size = len;
+}
 
-int SSL_CTX_set_read_ahead(SSL_CTX *ctx, int yes) { return 1; }
+void SSL_set_default_read_buffer_len(SSL *ssl, size_t len) {
```

Comment:
General comment, since we are adding new public functions, should we be checking whether parameters are null or not, and returning an error or aborting (if that is the only option)?